app/routes.py

```diff
@@ -1,2 +1,344 @@
+from app.models.goal import Goal
+from datetime import datetime
+import os
+import requests
+from dotenv import load_dotenv
+load_dotenv()
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST", "GET"])
+def handle_tasks():
+    if request.method == "GET":
+        task = request.args.get("task")
+        task_query = request.args.get("sort")
+
+        if task_query:
+            if task_query == "asc":
+                tasks = Task.query.order_by(Task.title.asc())
+            elif task_query == "desc":
+                tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append(
+                {
+                    "id" : task.task_id,
+                    "title" : task.title,
+                    "description" : task.description,
+                    "is_complete" : task.is_complete
+                }
+            )
+        
+        return jsonify(tasks_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        #replace with try/except
+        #this is my guard clause
+        if (("title" not in request_body.keys()) or 
+        ("description" not in request_body.keys()) or
+        ("completed_at" not in request_body.keys())):
+            return jsonify(
+                {"details": "Invalid data"}
+            ), 400
+
+        if not request_body["completed_at"]:
+            new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"],
+                            )
+            db.session.add(new_task)
+            db.session.commit()
+
+
+            return jsonify({"task":
+                {
+                "id" : new_task.task_id,
+                "title" : new_task.title,
+                "description" : new_task.description,
+                "is_complete" : new_task.is_complete
+                    }
+                }), 201
+
+        elif request_body["completed_at"]:
+            new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"],
+                            is_complete=True
+                            )
+            db.session.add(new_task)
+            db.session.commit()
+
+
+            return jsonify({"task":
+                {
+                "id" : new_task.task_id,
+                "title" : new_task.title,
+                "description" : new_task.description,
+                "is_complete" : new_task.is_complete
+                    }
+                }), 201
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
```

Comment:
Each of these verbs does depend on retrieving an existing task from the database by id, so there's a stronger case for keeping these together. But we could still think about splitting these into separate functions (e.g. `get_task`, `update_task`, and `delete_task`). We would need to either duplicate the lookup code in each function, or create a helper function that can be shared among the separated functions.